com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -803,7 +803,6 @@ public SocketTasks StartClient()
             }
 
             Initialize(false);
-            m_MessagingSystem.ClientConnected(ServerClientId);
```

Comment:
should you also remove the equivalent call in StartHost()?

Reply:
No, that one is needed since the host won't get a "connected" call from the server (since it is the server). But unlike this one, that one comes after the server socket's open, so it will get the correct server ID instead of assuming 0

Reply:
```
            var socketTasks = NetworkConfig.NetworkTransport.StartServer();
            m_MessagingSystem.ClientConnected(ServerClientId);
```
We do not wait until `socketTasks` is complete in the code above. You may/will run into the same issue. In `UnitryTransport` we luck out because currently the relevant connection will always have id `0`, but this (and the code which follows it) needs to be fixed eventually as well.